csharp/test/Ice/binding/RemoteObjectAdapterI.cs

```diff
@@ -14,20 +15,12 @@ public RemoteObjectAdapter(ObjectAdapter adapter)
         {
             _adapter = adapter;
             _testIntf = _adapter.Add("test", new TestIntf(), ITestIntfPrx.Factory);
-            _adapter.Activate();
         }
 
-        public ITestIntfPrx GetTestIntf(Current current, CancellationToken cancel) => _testIntf;
+        public ValueTask<ITestIntfPrx> GetTestIntfAsync(Current current, CancellationToken cancel) =>
+            new(_testIntf);
 
-        public void Deactivate(Current current, CancellationToken cancel)
-        {
-            try
-            {
-                _adapter.Dispose();
-            }
-            catch (ObjectDisposedException)
-            {
-            }
-        }
+        public async ValueTask DeactivateAsync(Current current, CancellationToken cancel) =>
+            await _adapter.DisposeAsync();
```

Comment:
Do we really need the async/await here? can't we just return the value task from `_adapter.DisposeAsync()`?

Reply:
Fixed, we don't need it.